src/python/pants/backend/python/docs/sphinx/rules.py

```diff
@@ -0,0 +1,64 @@
+# Copyright 2022 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from __future__ import annotations
+
+from dataclasses import dataclass
+
+from pants.backend.python.docs.sphinx.sphinx_subsystem import SphinxSubsystem
+from pants.backend.python.docs.sphinx.target_types import SphinxProjectSourcesField
+from pants.backend.python.util_rules import pex
+from pants.backend.python.util_rules.pex import PexRequest, VenvPex, VenvPexProcess
+from pants.core.goals.package import (
+    BuiltPackage,
+    BuiltPackageArtifact,
+    OutputPathField,
+    PackageFieldSet,
+)
+from pants.engine.internals.native_engine import AddPrefix, Digest, RemovePrefix
+from pants.engine.process import ProcessResult
+from pants.engine.rules import Get, MultiGet, collect_rules, rule
+from pants.engine.target import HydratedSources, HydrateSourcesRequest
+from pants.engine.unions import UnionRule
+from pants.util.logging import LogLevel
+
+
+@dataclass(frozen=True)
+class SphinxPackageFieldSet(PackageFieldSet):
+    required_fields = (SphinxProjectSourcesField,)
+
+    sources: SphinxProjectSourcesField
+    output_path: OutputPathField
+
+
+@rule
+async def generate_sphinx_docs(
+    field_set: SphinxPackageFieldSet, sphinx: SphinxSubsystem
+) -> BuiltPackage:
+    sources, sphinx_pex = await MultiGet(
+        Get(HydratedSources, HydrateSourcesRequest(field_set.sources)),
+        Get(VenvPex, PexRequest, sphinx.to_pex_request()),
+    )
+    result = await Get(
+        ProcessResult,
+        VenvPexProcess(
+            sphinx_pex,
+            argv=(field_set.address.spec_path or ".", "__build"),
```

Comment:
`__build` should probably be a constant. Or, since this is Python, a "constant".